device-connectors/src/testflinger_device_connectors/devices/maas2/maas_storage.py

```diff
@@ -32,14 +32,18 @@ class MaasStorageError(ProvisioningError):
 class MaasStorage:
     """Maas device connector storage module."""
 
-    def __init__(self, maas_user, node_id):
+    def __init__(self, maas_user, node_id, node_info=None):
         self.maas_user = maas_user
         self.node_id = node_id
         self.device_list = None
         self.init_data = None
-        self.node_info = self._node_read()
         self.block_ids = {}
         self.partition_sizes = {}
+        # Better support for testing
+        if node_info is None:
+            self.node_info = self._node_read()
+        else:
+            self.node_info = node_info
```

Comment:
can be a one-line assignment, but you can keep it as is since it's a personal preference
```suggestion
        # Better support for testing
        self.node_info = self._node_read() if node_info is None else node_info
```